lib/webflow/client.rb

```diff
@@ -131,17 +132,17 @@ def track_rate_limit(headers)
     # https://developers.webflow.com/#errors
     def self.error_codes
       {
-        [400,	SyntaxError] =>	'Request body was incorrectly formatted. Likely invalid JSON being sent up.',
-        [400,	InvalidAPIVersion] =>	'Requested an invalid API version',
-        [400,	UnsupportedVersion] =>	'Requested an API version that in unsupported by the requested route',
-        [400,	NotImplemented] =>	'This feature is not currently implemented',
-        [400,	ValidationError] =>	'Validation failure (see err field in the response)',
-        [400,	Conflict] =>	'Request has a conflict with existing data.',
-        [401,	Unauthorized] =>	'Provided access token is invalid or does not have access to requested resource',
-        [404,	NotFound] =>	'Requested resource not found',
-        [429,	RateLimit] =>	'The rate limit of the provided access_token has been reached. Please have your application respect the X-RateLimit-Remaining header we include on API responses.',
-        [500,	ServerError] =>	'We had a problem with our server. Try again later.',
-        [400,	UnknownError] =>	'An error occurred which is not enumerated here, but is not a server error.',
+        [400,  SyntaxError] =>  'Request body was incorrectly formatted. Likely invalid JSON being sent up.',
+        [400,  InvalidAPIVersion] =>  'Requested an invalid API version',
+        [400,  UnsupportedVersion] =>  'Requested an API version that in unsupported by the requested route',
+        [400,  NotImplemented] =>  'This feature is not currently implemented',
+        [400,  ValidationError] =>  'Validation failure (see err field in the response)',
+        [400,  Conflict] =>  'Request has a conflict with existing data.',
+        [401,  Unauthorized] =>  'Provided access token is invalid or does not have access to requested resource',
+        [404,  NotFound] =>  'Requested resource not found',
+        [429,  RateLimitError] =>  'The rate limit of the provided access_token has been reached. Please have your application respect the X-RateLimit-Remaining header we include on API responses.',
+        [500,  ServerError] =>  'We had a problem with our server. Try again later.',
+        [400,  UnknownError] =>  'An error occurred which is not enumerated here, but is not a server error.',
```

Comment:
i'm a bit surprised, that this does not raise any exceptions, while all the constants have never been initialized...

Reply:
the class method is never called so ruby is too lazy to evaluate it

Reply:
i think a proper way of handling this would be some kind of result object that can be asked for status:

```
result.ok?
response.items
respone.error
```

i would assume that all 4xx and 5xx responses can be treated as errors. the translation of error-codes to text, as kinda bogously implemented, would be a nice bonus.